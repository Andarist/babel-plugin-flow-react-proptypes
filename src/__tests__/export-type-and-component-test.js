var babel = require('babel-core');
var content = `
// @flow

import React from 'react';

export type T = {
  f: Function,
  i: number,
  x: 'foo' |Â 'baz',
};

const C = ({
  f,
}: T) => {
   <div></div>
};

export default C;
`;

it('export-type-and-component', () => {
  var res = babel.transform(content, {
    babelrc: false,
    presets: ['es2015', 'stage-1', 'react'],
    plugins: ['syntax-flow', require('../')],
  }).code;
  expect(res).toMatchSnapshot();
});
