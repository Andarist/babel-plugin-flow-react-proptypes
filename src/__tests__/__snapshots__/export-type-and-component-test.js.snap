// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`export-type-and-component 1`] = `
"'use strict';

Object.defineProperty(exports, \\"__esModule\\", {
  value: true
});

var _react = require('react');

var _react2 = _interopRequireDefault(_react);

var _propTypes = require('prop-types');

var _propTypes2 = _interopRequireDefault(_propTypes);

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

var babelPluginFlowReactPropTypes_proptype_T = {
  f: _propTypes2.default.func.isRequired,
  i: _propTypes2.default.number.isRequired,
  x: _propTypes2.default.oneOf(['foo', 'baz']).isRequired
};
if (typeof exports !== 'undefined') Object.defineProperty(exports, 'babelPluginFlowReactPropTypes_proptype_T', {
  value: babelPluginFlowReactPropTypes_proptype_T,
  configurable: true
});


var C = function C(_ref) {
  var f = _ref.f;

  _react2.default.createElement('div', null);
};

C.propTypes = babelPluginFlowReactPropTypes_proptype_T;
exports.default = C;"
`;
